examples/run_CPMG.py

```diff
@@ -10,125 +10,172 @@
+        # The way to be sure of this would be to capture on a scope and
+        # measure from one 180 to the next (or actually several, since
+        # this error would be cumulative
+        ("delay", config_dict["repetition_us"]),
+    ],
+    nScans=config_dict["nScans"],
+    indirect_idx=0,
+    indirect_len=1,
+    adcOffset=config_dict["adc_offset"],
+    carrierFreq_MHz=config_dict["carrierFreq_MHz"],
+    nPoints=nPoints,
+    acq_time_ms=config_dict["echo_acq_ms"],
+    SW_kHz=config_dict["SW_kHz"],
+    ret_data=None,
 )
-# }}}
-# {{{ chunk and save data
-if phase_cycling:
-    data.chunk("t", ["ph1", "echo", "t2"], [len(ph1_cyc), config_dict["nEchoes"], -1])
-    data.setaxis("ph1", ph1_cyc / 4)
-    data.setaxis("echo", r_[0 : config_dict["nEchoes"]])
-    if config_dict["nScans"] > 1:
-        data.setaxis("nScans", r_[0 : config_dict["nScans"]])
-    data.squeeze()
-    data.set_units("t2", "s")
-    fl.next("Raw - time")
-    fl.image(
-        data.C.mean("nScans"))
-    data.reorder("t2", first=False)
-    for_plot = data.C
-    for_plot.ft('t2',shift=True)
-    for_plot.ft(['ph1'], unitary = True)
-    fl.next('FTed data')
-    fl.image(for_plot.C.mean("nScans")
-    )
-else:
-    if config_dict["nScans"] > 1:
-        data.setaxis("nScans", r_[0 : config_dict["nScans"]])
-    data.rename("t","t2")
-    fl.next("Raw - time")
-    fl.image(
-        data.C.mean("nScans"))
-    data.reorder("t2", first=False)
-    for_plot = echo_data.C
-    for_plot.ft('t2',shift=True)
-    fl.next('FTed data')
-    fl.image(for_plot)
-data.name(config_dict["type"] + "_" + config_dict["cpmg_counter"])
-data.set_prop("postproc_type", "spincore_CPMGv2")
+## {{{ chunk and save data
+data.set_prop("postproc_type", "spincore_diffph_SE_v1")
+data.set_prop("coherence_pathway", {"ph_overall": -1, "ph1": +1})
 data.set_prop("acq_params", config_dict.asdict())
-target_directory = getDATADIR(exp_type="ODNP_NMR_comp/CPMG")
+# PR COMMENT: def of nodename was objectively convoluted -- I fix
+nodename = config_dict["type"] + "_" + str(config_dict["cpmg_counter"])
+data.name(nodename)
+data.chunk(
+    "t",
+    ["ph2", "ph_diff", "nEcho", "t2"],
+    [len(ph2), len(ph_diff), int(config_dict["nEchoes"]), -1],
+)
+# PR COMMENT: you are setting the axes multiple times
+data.labels(
+    {
+        "nEcho": r_[0 : int(config_dict["nEchoes"])],
+        "ph2": r_[0 : len(ph2)],
+        "ph_diff": r_[0 : len(ph_diff)],
+    }
+)
+data.setaxis("ph2", ph2 / 4)
+data.setaxis("ph_diff", ph_diff / 4)
+# }}}
 filename_out = filename + ".h5"
-nodename = data.name()
 if os.path.exists(f"{filename_out}"):
     print("this file already exists so we will add a node to it!")
     with h5py.File(
         os.path.normpath(os.path.join(target_directory, f"{filename_out}"))
     ) as fp:
         if nodename in fp.keys():
             print("this nodename already exists, so I will call it temp_cpmg")
+            # PR COMMENT: this is objectively not a good solution.  You
```

Comment:
address

Reply:
fixed based on combined_ODNP.py in examples